plugins/behaviour/versionable/src/Extension/Versionable.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Behaviour.versionable
+ *
+ * @copyright   (C) 2016 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Plugin\Behaviour\Versionable\Extension;
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Application\CMSApplicationInterface;
+use Joomla\CMS\Component\ComponentHelper;
+use Joomla\CMS\Event\Table\AfterStoreEvent;
+use Joomla\CMS\Event\Table\BeforeDeleteEvent;
+use Joomla\CMS\Helper\CMSHelper;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Versioning\VersionableTableInterface;
+use Joomla\CMS\Versioning\Versioning;
+use Joomla\Event\DispatcherInterface;
+use Joomla\Event\SubscriberInterface;
+use Joomla\Filter\InputFilter;
+
+/**
+ * Implements the Versionable behaviour which allows extensions to automatically support content history for their content items.
+ *
+ * This plugin supersedes JTableObserverContenthistory.
+ *
+ * @since  4.0.0
+ */
+class Versionable extends CMSPlugin implements SubscriberInterface
```

Comment:
Should the class be `final`?
```suggestion
final class Versionable extends CMSPlugin implements SubscriberInterface
```